### PR TITLE
Feat emojis and image uploads

### PR DESCRIPTION
**Pull request by SinchanaMS**
This request adds the following features:

1. Loaders and Toasts
2. Image uploads in profile for cover pic and avatar
3. Emoji picker, images/gif uploads in posts
4. Test login

### CODE REVIEW

File: src/features/user/Signup.jsx [hunk 1]
```diff
@@ -75,6 +77,7 @@ export default function Signup() {
           { merge: true }
         );
         localStorage.setItem("authToken", userAuth.user.accessToken);
+        toast.success(`Welcome to Jigsaw, ${name}!!`, { id: loader });
```
Comment (srejitk) [inline src/features/user/Signup.jsx:80]: Good work here! :)

File: src/firebase/firebase-calls.js [hunk 2]
```diff
@@ -21,104 +25,206 @@ const date = new Date().toLocaleDateString("en-IN", {
...
+      login({
+        email: userAuth.user.email,
+        uid: userAuth.user.uid,
+        displayName: userAuth.user.displayName,
+        photoURL: userAuth.user.photoURL,
+      })
+    );
+    localStorage.setItem("authToken", userAuth.user.accessToken);
+    toast.success(`Welcome back, ${userAuth.user.displayName}!`, {
+      id: loader,
```
Comment (srejitk) [inline src/firebase/firebase-calls.js:48]: Learnt this today! Great work!

**Review by srejitk — APPROVED** [review]
Didn't see major changes. Most were refactor. Good work with the hot toast and random cover pic.

**Comment by rutvikpumak** [comment]
Code looks good and well.
Functionality working as expected

**Review by dipen2000 — APPROVED** [review]
Really good work and the features are working correctly a small nit pick you gotta add cursor pointer on icons of image uploads and places like that. otherwise great work.

**Comment by SinchanaMS** [comment]
Thank you for the reviews! Merging to dev.

**Pull request merged**